server/market/orderrouter.go

```diff
@@ -229,13 +221,32 @@ func (r *OrderRouter) handleLimit(user account.AccountID, msg *msgjson.Message)
 		return msgjson.NewError(msgjson.OrderParameterError, "unknown time-in-force")
 	}
 
-	// Commitment.
+	// Commitment
 	if len(limit.Commit) != order.CommitmentSize {
 		return msgjson.NewError(msgjson.OrderParameterError, "invalid commitment")
 	}
 	var commit order.Commitment
 	copy(commit[:], limit.Commit)
 
+	fundingAsset := coins.funding
+
+	coinIDs := make([]order.CoinID, 0, len(limit.Trade.Coins))
+	coinStrs := make([]string, 0, len(limit.Trade.Coins))
+	for _, coin := range limit.Trade.Coins {
+		// Check that the outpoint isn't locked.
+		coinID := order.CoinID(coin.ID)
+		if tunnel.CoinLocked(coins.funding.ID, coinID) {
+			return msgjson.NewError(msgjson.FundingError, fmt.Sprintf("coin %v is locked", coinID))
+		}
+
+		coinStr, err := fundingAsset.Backend.ValidateCoinID(coinID)
+		if err != nil {
+			return msgjson.NewError(msgjson.FundingError, fmt.Sprintf("invalid coin ID %v: %v", coinID, err))
```

Comment:
This is implicitly validated in the coinwaiter, but I want the decoded coin strings, and it's good to do it straight away.